### PR TITLE
Release/0.1.0.alpha1

### PR DESCRIPTION
**Pull request by engarpe**
🎉 **Alpha1 release of the data program builder SDK**

## 🛠 Changes
- Adds DataProgram, Project, Worker, and Task interfaces
- Add DataProgram mixin client operation
- Stores environment configurations in local `~/.superai/.secrets.yaml` file to easy development

## 🧪 How is this tested?
- `/tests` folder

## 🗒️ Release notes

🎉  **Alpha 1 release of the Data Program Builder SDK**

Introducing the super.AI Data Program Builder! Our new SDK lets you harness the power and flexibility of schema to build custom Data Programs. Create the perfect tool to handle the complexity and specifics of your use case. Here are a few of the things we've packed into this first release:
    
- Choose your input and output data types
- Set custom parameters
    - Let users define their own labels and instructions when designing a project
- Create your own workflows
    - Split large files into chunks
    - Choose where your input data gets labeled
    - Chain smaller labeling tasks together to reduce complexity
- [Read the documentation](https://super.ai/docs/data-program-builder-overview) to get started, learn how to add workflows and tasks, and view complete reference material on all the data types we offer, alongside their corresponding task schema functions and JSON schema definitions—everything you need to get building

[Email us](mailto:dataprogrammer@super.ai?subject=I'd%20like%20to%20be%20a%20data%20programmer!&body=My%20super.AI%20account%20email%20is...) to set up a data programmer account, then follow our [installation instructions](https://super.ai/docs/data-program-builder-overview#requirements).

Read our [two-part getting started guide](https://super.ai/docs/getting-started-with-the-superai-data-program-builder-part-one) to learn the basics.

### User-facing change?
- [ ] No. Skip the rest of this section
- [x] Yes. Give a description of this change to be included in the release notes.

### What component(s) does this PR affect?

**Components**
- [x] CLI
- [x] APIs

**How should the PR be classified in the release notes? Choose one:**
- [x] `breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `bug`-fix - A user-facing bug fix worth mentioning in the release notes
- [ ] `documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,4 +97,104 @@ client.create_jobs(
...
+  1. Create a virtualenv (we recommend to use [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv) or conda)
+  2. Run `pip install --upgrade pip==20.1.1`
+  3. Run `pip install "awscli"`
+  4. Run `pip install --upgrade "superai>=0.1.0.a1"`
+  5. Run `superai env set -e sandbox` 
+  6. Run `superai login -u <user_email>`
+  7. Verify that the pip was configured correctly by running `pip config get global.index-url`
+     If the response is empty then run `superai login --show-pip -u <user_email>` and copy/paste the 
+     `pip config set...` command as indicated
+  8. Install superai in dataprogramming mode `pip install --upgrade "superai[dp]>=0.1.0.alpha1"`
```
Comment (lpfann) [inline README.md:118]: should this be `a1` at the end?

Comment (engarpe) [reply README.md:118]: Yes, that is the version being released

File: superai/apis/project.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
+from __future__ import absolute_import
+
+import logging
+import re  # noqa: F401
+from abc import ABC, abstractmethod
+
+# python 2 and python 3 compatibility library
```
Comment (lpfann) [inline superai/apis/project.py:7]: Still includes Python2 code, but is not a blocker. Btw. where is `six` dependency declared? If its coming from a third party dependency, maybe we should specify it until removed.

File: superai/data_program/protocol/task_helpers.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    task_inputs,
+    task_outputs,
+    qualifications,
+    task_price,
+    task_expiry_time,
+    n_resend,
+    task_name,
+    excluded_ids=None,
+):
+    from canotic.hatchery import turbine_api as tb  # TODO: Remove dependency
```
Comment (lpfann) [inline superai/data_program/protocol/task_helpers.py:71]: Still depending on `canotic`

**Review by lpfann — CHANGES_REQUESTED** [review]

File: superai/data_program/protocol/task_helpers.py (hunk 3 above)
Comment (engarpe) [reply superai/data_program/protocol/task_helpers.py:71]: Yes, this will be removed in 0.1.0 release

File: README.md (hunk 1 above)
Comment (lpfann) [reply README.md:118]: Installation step 4. uses `a1` and step 8. uses `alpha1`. I think it should be consistent.

**Review by ankitmishra262 — APPROVED** [review]

**Review by lpfann — APPROVED** [review]

File: superai/apis/project.py (hunk 2 above)
Comment (engarpe) [reply superai/apis/project.py:7]: I added the dependency

**Pull request merged**